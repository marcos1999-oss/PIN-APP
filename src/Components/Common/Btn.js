import React from "react";

import { Text, View, ActivityIndicator } from "react-native";

import { Button } from "native-base";
import { moderateScale } from "../../Utils/scaling";

export default (props) => {
  return (
    <Button
      {...props}
      style={[
        { opacity: 1 },
        props.style,
        props.color && { backgroundColor: props.color },
        props.disabled && { opacity: 0.9 },
      ]}
    >
      {props.isLoading && <ActivityIndicator size="large" color={props.loadingColor || "#fff"} />}
      {!props.isLoading && (
        <React.Fragment>
          {props.preText && (
            <Text style={props.preTextStyle}>{props.preText} </Text>
          )}
          <Text
            style={[
              { fontSize: moderateScale(16) },
              props.textStyle,
              props.textColor && { color: props.textColor },
            ]}
          >
            {props.title && props.title}
          </Text>
        </React.Fragment>
      )}
    </Button>
  );
};
