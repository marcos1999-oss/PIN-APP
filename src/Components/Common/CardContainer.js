import React from "react";
import { Card, CardItem, Body } from "native-base";
import { verticalScale } from "../../Utils/scaling";

export default (props) => (
  <Card style={[{ borderRadius: verticalScale(20) }, props.style]}>
    <CardItem style={{ borderRadius: verticalScale(20) }}>
      <Body>{props.children}</Body>
    </CardItem>
  </Card>
);
