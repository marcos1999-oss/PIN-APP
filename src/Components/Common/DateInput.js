import React from "react";
import { Text } from "react-native";
import { Item } from "native-base";
import DatePicker from "react-native-datepicker";
import style from "./style/input";
import { moderateScale, verticalScale, scale } from "../../Utils/scaling";

const customStyles = {
  placeholderText: {
    fontSize: moderateScale(12),
    color: "#000",
  },
  dateText: {
    fontSize: moderateScale(12),
    color: "#000",
  },
  dateInput: {
    borderColor: "rgba(255,255,255,0.4)",
    borderWidth: 0,
    borderBottomWidth: 1,
    alignItems: "flex-start",
    paddingVertical: verticalScale(5),
    paddingHorizontal: scale(5),
  },
};

export default ({ form, formKey, ref, setRef, containerStyle, ...rest }) => {
  return (
    <Item style={containerStyle}>
      <DatePicker
        mode="date"
        // date={form.values.dob}
        date={form.values[formKey.toString()]}
        format="DD MMMM YYYY"
        showIcon={false}
        style={{ flexGrow: 1, ...rest.style }}
        confirmBtnText="Confirm"
        cancelBtnText="Cancel"
        customStyles={{ ...customStyles, ...rest.customStyles }}
        onDateChange={(date) => {
          form.setFieldValue(formKey, date);
        }}
        {...rest}
      />
      <Text style={style.m_error_msg}>{form.errors[formKey.toString()]}</Text>
    </Item>
  );
};
