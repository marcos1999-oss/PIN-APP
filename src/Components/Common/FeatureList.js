import React from 'react'
import { View, Image } from 'react-native'
import { Text } from 'native-base'
import { isUndefined, find } from 'lodash'

import {
  alcohol,
  baby,
  driveThrough,
  heart,
  music,
  parking,
  party,
  pet,
  reservation,
  wifi,
  valetParking,
} from '../../Assets'


const features = [
  { name: 'Alcohol', iconName: 'alcohol', icon: alcohol },
  { name: 'Baby Friendly', iconName: 'baby', icon: baby },
  { name: 'Drive Thru', iconName: 'driveThrough', icon: driveThrough },
  { name: 'Romantic', iconName: 'heart', icon: heart },
  { name: 'Music', iconName: 'music', icon: music },
  { name: 'Parking', iconName: 'parking', icon: parking },
  { name: 'Parties & Group', iconName: 'party', icon: party },
  { name: 'Pet Friendly', iconName: 'pet', icon: pet },
  { name: 'Reservations', iconName: 'reservation', icon: reservation },
  { name: 'Wifi', iconName: 'wifi', icon: wifi },
  { name: 'Valet Parking', iconName: 'valetParking', icon: valetParking },
];

const getFeature = (feature) => {
  return find(features, { iconName: feature.icon });
};

const renderFeatures = (features, style, imageStyle) => {
  return features.map((feature) => {
    const validFeature = getFeature(feature);

    if (isUndefined(validFeature)) {
      return (<View key={`renderFeatures-${feature.id}`}></View>);
    } else {
      return (
        <View key={ feature.id } style={ style }>
          <Image source={ validFeature.icon } style={ imageStyle } />
        </View>
      );
    }
  });
};

const renderDetailedFeatures = (features, containerStyle, imageContainerStyle, imageStyle, textContainerStyle, textStyle) => {
  return features.map((feature) => {
    const validFeature = getFeature(feature);

    if (isUndefined(validFeature)) {
      return (<View key={`renderDetailedFeatures-${feature.id}`}></View>);
    } else {
      return (
        <View style={ containerStyle } key={ feature.id }>
          <View style={ imageContainerStyle }>
            <Image source={ validFeature.icon } style={ imageStyle } />
          </View>

          <View style={ textContainerStyle }>
            <Text style={ textStyle }>{ feature.name }</Text>
          </View>
        </View>
      );
    }
  });
};

export const FeatureList = ({ features, containerStyle, style, imageStyle }) => (
  <View style={ containerStyle }>
    { renderFeatures(features, style, imageStyle) }
  </View>
);

export const DetailedFeatureList = ({ features, outerContainerStyle, containerStyle, imageContainerStyle, imageStyle, textContainerStyle, textStyle }) => (
  <View style={ outerContainerStyle }>
    { renderDetailedFeatures(features, containerStyle, imageContainerStyle, imageStyle, textContainerStyle, textStyle) }
  </View>
);
