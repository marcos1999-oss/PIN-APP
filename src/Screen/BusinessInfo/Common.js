import React from "react";
import { View, Dimensions, Text } from "react-native";
import { Card, CardItem, Body } from "native-base";
import Image from "react-native-auto-height-image";
import { city2 } from "../../Assets";

/*Style*/
import style from "./style.js";
import { scale, verticalScale } from "../../Utils/scaling";

export const FormContainer = props => (
  <View>
    <View style={{ alignItems: "center", zIndex: 2 }}>
      <Image source={city2} width={Dimensions.get("window").width / 1.5} />
    </View>
    <View
      style={{
        paddingHorizontal: scale(10),
        marginTop: verticalScale(-10),
        zIndex: 1,
      }}
    >
      <Card style={{ borderRadius: verticalScale(20) }}>
        <CardItem style={{ borderRadius: verticalScale(20) }}>
          <Body>
            <View
              style={{
                paddingHorizontal: scale(10),
                paddingVertical: verticalScale(10),
                width: "100%",
              }}
            >
              {props.children}
            </View>
          </Body>
        </CardItem>
      </Card>
    </View>
  </View>
);
