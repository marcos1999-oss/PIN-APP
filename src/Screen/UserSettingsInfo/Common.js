import React from "react";
import { View } from "react-native";
import { Card, CardItem, Body } from "native-base";
import { scale, verticalScale } from "../../Utils/scaling";

export const FormContainer = (props) => (
  <View>
    <View
      style={{
        paddingHorizontal: scale(10),
        marginTop: verticalScale(-10),
        zIndex: 1,
      }}
    >
      <Card style={{ borderRadius: verticalScale(20) }}>
        <CardItem style={{ borderRadius: verticalScale(20) }}>
          <Body>
            <View
              style={{
                paddingHorizontal: scale(10),
                paddingVertical: verticalScale(10),
                width: "100%",
              }}
            >
              {props.children}
            </View>
          </Body>
        </CardItem>
      </Card>
    </View>
  </View>
);
