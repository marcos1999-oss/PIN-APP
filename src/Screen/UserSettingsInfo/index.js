import UserSettingsUsername from "./UserSettingsUsername";
import UserSettingsName from "./UserSettingsName";
import UserSettingsEmail from "./UserSettingsEmail";
import UserSettingsPhone from "./UserSettingsPhone";
import UserSettingsBirthday from "./UserSettingsBirthday";
import ForgotPassword from "./ForgotPassword";
import ForgetPasswordSuccess from "./ForgetPasswordSuccess";

export {
  UserSettingsUsername,
  UserSettingsName,
  UserSettingsEmail,
  UserSettingsPhone,
  UserSettingsBirthday,
    ForgotPassword,ForgetPasswordSuccess
};
