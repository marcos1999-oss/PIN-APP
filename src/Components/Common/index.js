import React from "react";

import Btn from "./Btn";
import ScreenHeader from "./ScreenHeader";
import DismissKeyboard from "./DismissKeyboard";
import TextInput from "./TextInput";
import DateInput from "./DateInput";
import ScreenLoading from "./ScreenLoading";
import ListLink, { ListLinkHeader } from "./ListLink";
import CardContainer from "./CardContainer";
import { FeatureList, DetailedFeatureList } from './FeatureList'

export {
  Btn,
  ScreenHeader,
  DismissKeyboard,
  TextInput,
  DateInput,
  ScreenLoading,
  ListLink,
  ListLinkHeader,
  CardContainer,
  FeatureList,
  DetailedFeatureList,
};
