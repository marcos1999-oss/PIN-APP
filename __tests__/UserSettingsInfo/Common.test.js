/**
 * @format
 * @lint-ignore-every XPLATJSCOPYRIGHT1
 */

import {Text} from 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

import {FormContainer} from '../../src/Screen/UserSettingsInfo/Common';

jest.mock('react-native-cached-images', () => {
  return {
    DocumentDir: () => {},
    ImageCache: {
      get: {
        clear: () => {},
      },
    },
  }
});

test('renders correctly', () => {
  const tree = renderer.create(<FormContainer><Text>Test</Text></FormContainer>).toJSON();
  expect(tree).toMatchSnapshot();
});
