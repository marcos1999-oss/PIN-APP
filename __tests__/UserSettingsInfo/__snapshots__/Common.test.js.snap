// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "marginTop": -20,
        "paddingHorizontal": 20,
        "zIndex": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#fff",
            "borderBottomWidth": 0.5,
            "borderColor": "#ccc",
            "borderLeftWidth": 0.5,
            "borderRadius": 2,
            "borderRightWidth": 0.5,
            "borderTopWidth": 0.5,
            "elevation": 3,
            "flexWrap": "nowrap",
            "marginBottom": 5,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 5,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.1,
            "shadowRadius": 1.5,
          },
          Object {
            "borderRadius": 40,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#fff",
              "borderRadius": 2,
              "flexDirection": "row",
              "paddingBottom": 10,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 10,
            },
            Object {
              "borderRadius": 40,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "alignSelf": "stretch",
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "paddingHorizontal": 20,
                "paddingVertical": 20,
                "width": "100%",
              }
            }
          >
            <Text>
              Test
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
